CONTRIBUTING.md

```diff
@@ -26,133 +26,169 @@ We specifically do not merge PRs **without prior issues** that:
+
+---
+
+### Updating the changelog
+
+We are keeping track of the changes to OpenTofu in the [CHANGELOG.md](CHANGELOG.md) file. Please update it when you add features or fix bugs in OpeTofu.
+
+---
+
+### Signing off your commits
+
+When you contribute code to OpenTofu, we require you to add a [Developer Certificate of Origin](https://developercertificate.org/) sign-off. Please read the DCO carefully before you proceed and only contribute code you have written yourself. Please do not add code that you have not written from scratch yourself without discussing it in the related issue first.
+
+The simplest way to add a sign-off is to use the `-s` command when you commit:
+
 ```
-~> git commit -s -m 'This is my commit message'
+git commit -s -m "My commit message"
 ```
 
-You can find more details about the DCO checker in the [DCO app repo](https://github.com/dcoapp/app).
+> [!IMPORTANT]
+> Make sure your `user.name` and `user.email` setting in Git matches your GitHub settings. This will allow the automated DCO check to pass and avoid delays when merging your PR.
 
-Additionally, please update [the changelog](CHANGELOG.md) if you're making any user-facing changes.
+> [!TIP]
+> Have you forgotten your sign-off? You can use `git rebase --signoff` to add it after the fact.
 
-## Working on the Code
+---
 
-If you wish to work on the OpenTofu CLI source code, you'll first need to install the [Git](https://git-scm.com/) version control system. Use Git to clone this repository into a location of your choice. OpenTofu uses [Go Modules](https://blog.golang.org/using-go-modules), and so you should *not* clone it inside your `GOPATH`.
+### A note on copyright
 
-After that, you can either install the [Go](https://golang.org/) compiler locally, or use [Docker](https://www.docker.com/) to build in a container. At this time the OpenTofu development environment targets only Linux and MacOS systems. While OpenTofu itself is compatible with Windows, unfortunately the unit test suite currently contains Unix-specific assumptions around maximum path lengths, path separators, etc. This means that using Docker is the best option if working on Windows.
+We take copyright and intellectual property very seriously. A few quick rules should help you:
 
-If using Visual Studio Code or Goland/IntelliJ, a [devcontainer](.devcontainer.json) is included which integrates directly with Docker. In Visual Studio Code, you can install the [Remote Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extension, then reopen the project to get a prompt about activating the devcontainer. In Goland/Intellij, open the `.devcontainers.json` file and click the purple cube icon that appears next to the line numbers to activate the dev container. At this point you can proceed as if [building natively](#building-natively) on Linux.
+1. You should write most, if not all of your code yourself.
```

Comment:
Can we start with the note about LLMs, and then put this point in the context of that? Otherwise it's a bit unclear - does it mean I can't submit my friend's code? Or does it refer to not copying from the legacy repo?

Reply:
Technically no, you shouldn't submit your friends code without their permission. This may need rephrasing.

Reply:
Rephrased.